src/modules/handlers/screen/poster/gewis/gewis-poster-screen-controller.ts

```diff
@@ -0,0 +1,103 @@
+import {BasePosterResponse, BasePosterScreenController, BorrelModeParams} from "../base-poster-screen-controller";
+import HandlerManager from "../../../../root/handler-manager";
+import {Screen} from "../../../../root/entities";
+import {GewisPosterScreenHandler} from "../../index";
+import {Body, Get, Post, Put, Query, Request, Route, Security, Tags} from "tsoa";
+import {SecurityNames} from "../../../../../helpers/security";
+import {securityGroups} from "../../../../../helpers/security-groups";
+import logger from "../../../../../logger";
+import {Request as ExpressRequest} from "express";
+import {TrainResponse} from "../ns-trains-service";
+import GEWISPosterService, {GEWISPhotoAlbumParams} from "./gewis-poster-service";
+import OlympicsService from "../olympics-service";
+import {FeatureEnabled} from "../../../../server-settings";
+
+interface GewisPosterResponse extends BasePosterResponse {
+    borrelMode: boolean;
+}
+
+@Route('handler/screen/gewis-poster')
+@Tags('Handlers')
+@FeatureEnabled('GewisPosterScreenHandler')
+export class GewisPosterScreenController extends BasePosterScreenController {
+    protected screenHandler: GewisPosterScreenHandler;
+
+    constructor() {
+        super();
+        this.screenHandler = HandlerManager.getInstance()
+            .getHandlers(Screen)
+            .filter((h) => h.constructor.name === GewisPosterScreenHandler.name)[0] as GewisPosterScreenHandler;
+    }
+
+    @Security(SecurityNames.LOCAL, securityGroups.poster.base)
+    @Get('')
+    public async getGewisPosters(@Query() alwaysReturnBorrelPosters?: boolean): Promise<GewisPosterResponse> {
+        if (!this.screenHandler.posterManager.posters) {
+            try {
+                await this.screenHandler.posterManager.fetchPosters();
+            } catch (e) {
+                logger.error(e);
+            }
+        }
+        const posters = this.screenHandler.posterManager.posters ?? [];
+        if (alwaysReturnBorrelPosters || this.screenHandler.borrelMode) {
+            return {
+                posters,
+                borrelMode: this.screenHandler.borrelMode,
+            };
+        }
+        return {
+            posters: posters.filter((p) => !p.borrelMode),
+            borrelMode: false,
+        };
```

Comment:
Can't you (partly) replace this implementation with `super.getPosters()`? You seem to have implemented it there as well.

Reply:
Yeah, was a bit lazy with this :upside_down_face: